src/api/routes/awala.routes.ts

```diff
@@ -0,0 +1,96 @@
+import type { RouteOptions } from 'fastify';
+import Ajv from 'ajv';
+import addFormats from 'ajv-formats';
+import type { JSONSchema, FromSchema } from 'json-schema-to-ts';
+
+import { HTTP_STATUS_CODES } from '../../utilities/http.js';
+import type { PluginDone } from '../../utilities/fastify/PluginDone.js';
+import type { FastifyTypedInstance } from '../../utilities/fastify/FastifyTypedInstance.js';
+import { MEMBER_BUNDLE_REQUEST_SCHEMA } from '../../schemas/awala.schema.js';
+import { AwalaProblemType } from '../../AwalaProblemType.js';
+import { createMemberBundleRequest } from '../../awala.js';
+import type { ServiceOptions } from '../../serviceTypes.js';
+
+const ajv = addFormats(new Ajv());
+
+type ValidationResult<Schema extends JSONSchema> = FromSchema<Schema> | string;
```

Comment:
Hate to do this but there is something wrong with Typescript if I do `Result<FromSchema<Schema> , string>` it throws an error
<img width="944" alt="Screenshot 2023-04-17 at 15 43 07" src="https://user-images.githubusercontent.com/223112/232474558-3b5303a4-fbfc-41d7-a379-703b3d53c0e0.png">

Reply:
The change to `Result<FromSchema<Schema> , string>` makes sense to me and I can't see anything wrong with your code.

An option we considered last week was to throw a custom error here (custom as in a subclass of `Error`)? That'd get around this issue -- it'd be a pragmatic solution, certainly not the nicest one.

I'm just thinking this is a tiny function that won't be exposed outside this file and we've already spent a lot time trying to find a nice solution for it (first with the assertions), so at this point we should just go for something that works.